### ISSUE TITLE
List all the credential groups

### ISSUE THREAD

**Issue by vplasencia** [bandada-infra/bandada#323]
In the Bandada groups page, in the dashboard, list all the credential groups after My groups, so other people can use these credential groups.

The idea is that people don't have to prove that they meet the same requirement several times. If they are already part of a credential group, they don't have to prove the same for other group with the same credential.

**Comment by family36** [comment]
hi, @vplasencia 
I'd like to work on this.

**Comment by vplasencia** [comment]
Hey @family36! 

I just assigned the issue to you. Thank you very much! Feel free to ask any questions.

**Comment by family36** [comment]
hi, @vplasencia

I don't fully understand the intention of this issue:

"In the Bandada groups page, in the dashboard, list all the credential groups after My groups, so other people can use these credential groups."

・What specific operations does "use" refer to in this context?
・If we list all credential groups, what specifically becomes possible?

Could you please clarify these points?

**Comment by vplasencia** [comment]
Hey @family36!

Since the credential groups are public, it can be nice to show all the current credential groups so that people can reuse groups, and they don't have to create new ones if they can reuse the ones that have been created before. 

Because you can join a credential group if you meet the group criteria even if you were not invited, anyone that meets the criteria can join the group. 

For example, if you create a group for people that have contributed to a specific GitHub repository, then if someone needs to create the same group, they can just reuse yours.

Does it make sense?

The idea is to show all the credential groups after the `My Groups` section that you can see after you log in.

Then, there will be a section, called something like `All Credential Groups` where people will see all the credential groups that people have created.

**Comment by family36** [comment]
hi, @vplasencia

Thank you very much. I now understand the intent of the issue.

I have completed the implementation up to the point of displaying all credential groups. I have ensured that groups with a null credential and the ones I created myself are not displayed.

Does this implementation sufficiently meet the requirements of the issue?

![image](https://github.com/user-attachments/assets/64ad2d08-c472-433d-9d7a-00270dd38a33)

**Comment by vplasencia** [comment]
Hey @family36! It looks great 🤩

I think it's better to also include the ones you created, even if they were listed before. Because filters and labels can be included in the future, and it can be useful to have them all together. For example, there can be a filter to get all the groups that have GitHub followers as part of the credentials. 

Then the `All credentials groups` section will have all the credential groups (including the credential groups you created). 

What do you think?

**Comment by family36** [comment]
hi, @vplasencia

I have modified the code to include the groups I created in 'All credentials groups'.
May I create a PR for this?

Separately, I noticed that the layout breaks when the group name is very long, as shown in the image. Please disregard this if an issue has already been created for it.

![image](https://github.com/user-attachments/assets/acf54f8e-53ab-4006-92d3-d3490be9772e)

**Comment by vplasencia** [comment]
Hey @family36! Thank you very much. I looks great. Yes, feel free to create a PR for this issue.

> Separately, I noticed that the layout breaks when the group name is very long, as shown in the image. Please disregard this if an issue has already been created for it.

Good catch! Feel free to also create a new issue and a new PR for this. Let me know if you would like to work on it too so that when you create the issue I can assign it to you.

**Comment by vplasencia** [comment]
Hey @family36! I just created this issue as a general issue to make the dashboard app responsive because there are other components that are not responsive either. If you create the issue you found, I can add it to the one I created as a subtask so that it's easier to track and manage the list of components that are not responsive yet: https://github.com/bandada-infra/bandada/issues/551

**Comment by family36** [comment]
hi, @vplasencia

I have created a PR.

Thank you for creating the issue.

**Comment by vplasencia** [comment]
Hey @family36 I just merged the PR. Thank you very much! 

Regarding the other issue you found related to the group cards breaking the layout, I created this new issue: https://github.com/bandada-infra/bandada/issues/553

And added it as a task to the main issue related to making the dashboard app responsive: https://github.com/bandada-infra/bandada/issues/551 to keep track of all the tasks related to making the app responsive because there are other components that are not responsive yet.
